lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -403,6 +407,17 @@ collectOpsToPipeline(scf::ForOp forOp,
       // fixed.
       if (!loadInfo.sharedEncoding)
         continue;
+    } else if (isa<tt::LoadOp>(distAndUse.second)) {
```

Comment:
I think there is still one problem left even after this fix:
We calculate `stagesBetweenLoads` based on the longest chain that is present in `loadOpToDistAndUse`. If we can "break the chain" after `loadOpToDistAndUse` is filled, this value may be incorrect in some cases.
This being said, this fix will help most cases, and does not really make the situation worse, we already can have improper `loadOpToDistAndUse` because we are dropping the loads here.

Reply:
I see. Thanks for the explanation, @pawelszczerbuk !